lib/ruby_lsp/requests/document_symbol.rb

```diff
@@ -220,6 +248,33 @@ def create_document_symbol(name:, kind:, range_node:, selection_range_node:)
 
         symbol
       end
+
+      sig { params(node: SyntaxTree::Command).void }
+      def parse_test(node)
+        return unless node.block
+
+        argument = node.arguments.parts.first
+        return unless argument.is_a?(SyntaxTree::StringLiteral) && node.arguments.parts.size == 1
+
+        name_node = argument.parts.first
+        return unless name_node
+
+        full_name = argument.parts.map do |part|
+          case part
+          when SyntaxTree::TStringContent
+            part.value
+          when SyntaxTree::StringEmbExpr
+            "{ }"
+          end
+        end.join("")
+
+        create_document_symbol(
+          name: full_name,
```

Comment:
If we are doing special treatment for AS::TestCase `test` method calls, then I think we should also do the name transformation it does as well: https://github.com/rails/rails/blob/7c70791470fc517deb7c640bead9f1b47efb5539/activesupport/lib/active_support/testing/declarative.rb#L14

Reply:
Ok, so this was the point I was trying to make initially about runtime method generation and how we are skirting on the border of something that doesn't cleanly belong in the current implementation.

The way I see it is that this feature is implementing support for syntax that does metaprogramming to create methods at runtime. All we are doing is to expose those methods as they would be represented at runtime.

If we are doing that, then we should be doing it faithfully, i.e. show the method with the name it would have at runtime.

The concern about describe, etc blocks is not important right now, since each testing framework will need special handling for this anyway.